src/richie/apps/search/indexers/courses.py

```diff
@@ -380,27 +381,70 @@ def get_es_document_for_course(cls, course, index=None, action="index"):
                 if picture_info:
                     cover_images[language] = picture_info
 
-        # Prepare the related category icon
-        icon_images = {}
-        for plugin_model in CategoryPluginModel.objects.filter(
-            cmsplugin_ptr__placeholder__page=course.extended_object_id,
-            cmsplugin_ptr__placeholder__slot="course_icons",
-            cmsplugin_ptr__position=0,
-        ):
-            language = plugin_model.language
+        # Helper to shape relevant category data from a category page
+        def get_category_data(category_page):
+            # Get the parent & meta categories for the current category
+            meta_category = category_page.category.get_meta_category()
+            parent_category_page = category_page.get_parent_page()
+            # Nil the parent if it's the meta, as topmost category does not really have a parent
+            parent_category_page = (
+                parent_category_page
+                if parent_category_page.reverse_id
+                != meta_category.extended_object.reverse_id
+                else None
+            )
+
+            category_data = {
+                "name": category_page.reverse_id,
+                "meta_name": meta_category.extended_object.reverse_id,
+                "meta_title": {
+                    title["language"]: title["title"]
+                    for title in meta_category.extended_object.title_set.values()
+                },
+                "parent_name": None,
+                "parent_title": None,
+                "title": {
+                    title["language"]: title["title"]
+                    for title in category_page.title_set.values()
+                },
+            }
+
+            if parent_category_page:
+                category_data["parent_name"] = parent_category_page.reverse_id
+                category_data["parent_title"] = {
+                    title["language"]: title["title"]
+                    for title in parent_category_page.title_set.values()
+                }
+
+            return category_data
+
+        # Generate categories data as an OrderedDict so we can deduplicate the categories by pk
+        # (use as keys) while keeping them in the same order
+        categories_data = OrderedDict()
+        for category in course.get_categories():
```

Comment:
I think we need the category data to be available on the course model as well so that we can do the same thing on the DjangoCMS glimpse as we do in the React glimpse.

And maybe it is interesting to be able to get a category's data from the category model too?

Reply:
You're right about the course model.

For the category model, placeholder info is irrelevant there I think. As for the meta & parent info, it's not too hard to get from a category instance right now.

Reply:
seems legit